Index: lib/vendor/symfony/lib/helper/PartialHelper.php
===================================================================
--- lib/vendor/symfony/lib/helper/PartialHelper.php	(revision 7768)
+++ lib/vendor/symfony/lib/helper/PartialHelper.php	(working copy)
@@ -64,7 +64,7 @@
 
   if ($componentSlot = $viewInstance->getComponentSlot($name))
   {
-    return get_component($componentSlot[0], $componentSlot[1], $vars);
+    return get_component($componentSlot[0], $componentSlot[1], array_merge($componentSlot[2], $vars));
   }
 }
 
Index: lib/vendor/symfony/lib/view/sfView.class.php
===================================================================
--- lib/vendor/symfony/lib/view/sfView.class.php	(revision 7768)
+++ lib/vendor/symfony/lib/view/sfView.class.php	(working copy)
@@ -407,11 +412,12 @@
    * @param string A module name
    * @param string A component name
    */
-  public function setComponentSlot($attributeName, $moduleName, $componentName)
+  public function setComponentSlot($attributeName, $moduleName, $componentName, $vars = array())
   {
     $this->componentSlots[$attributeName]                   = array();
     $this->componentSlots[$attributeName]['module_name']    = $moduleName;
     $this->componentSlots[$attributeName]['component_name'] = $componentName;
+    $this->componentSlots[$attributeName]['vars']           = $vars;
   }
 
   /**
@@ -437,7 +443,7 @@
   {
     if (isset($this->componentSlots[$name]) && $this->componentSlots[$name]['module_name'] && $this->componentSlots[$name]['component_name'])
     {
-      return array($this->componentSlots[$name]['module_name'], $this->componentSlots[$name]['component_name']);
+      return array($this->componentSlots[$name]['module_name'], $this->componentSlots[$name]['component_name'], $this->componentSlots[$name]['vars']);
     }
 
     return null;
