Index: lib/vendor/symfony/lib/util/sfContext.class.php
===================================================================
--- lib/vendor/symfony/lib/util/sfContext.class.php	(revision 7938)
+++ lib/vendor/symfony/lib/util/sfContext.class.php	(working copy)
@@ -269,7 +269,7 @@
 
   public function retrieveObjects($class, $peerMethod)
   {
-    $retrievingClass = 'sf'.ucfirst(sfConfig::get('sf_orm', 'propel')).'DataRetriever';
+    $retrievingClass = ucfirst(sfConfig::get('sf_orm', 'propel')).'DataRetriever';
 
     return call_user_func(array($retrievingClass, 'retrieveObjects'), $class, $peerMethod);
   }
