Index: lib/vendor/symfony/lib/plugins/sfPropelPlugin/lib/task/sfPropelBuildAllTask.class.php
===================================================================
--- lib/vendor/symfony/lib/plugins/sfPropelPlugin/lib/task/sfPropelBuildAllTask.class.php	(revision 9414)
+++ lib/vendor/symfony/lib/plugins/sfPropelPlugin/lib/task/sfPropelBuildAllTask.class.php	(working copy)
@@ -63,13 +63,6 @@
     $buildSql->setCommandApplication($this->commandApplication);
     $buildSql->run();
 
-    if (!$options['skip-forms'])
-    {
-      $buildForms = new sfPropelBuildFormsTask($this->dispatcher, $this->formatter);
-      $buildForms->setCommandApplication($this->commandApplication);
-      $buildForms->run();
-    }
-
     $insertSql = new sfPropelInsertSqlTask($this->dispatcher, $this->formatter);
     $insertSql->setCommandApplication($this->commandApplication);
     $insertSql->run();
