Index: lib/vendor/symfony/lib/plugins/sfPropelPlugin/lib/vendor/propel-generator/classes/propel/engine/database/model/ForeignKey.php
===================================================================
--- lib/vendor/symfony/lib/plugins/sfPropelPlugin/lib/vendor/propel-generator/classes/propel/engine/database/model/ForeignKey.php	(revision 7888)
+++ lib/vendor/symfony/lib/plugins/sfPropelPlugin/lib/vendor/propel-generator/classes/propel/engine/database/model/ForeignKey.php	(working copy)
@@ -144,7 +144,7 @@
 	 */
 	public function getForeignTableName()
 	{
-		return $this->foreignTableName;
+		return 'q_'.$this->foreignTableName;
 	}
 
 	/**
Index: lib/vendor/symfony/lib/plugins/sfPropelPlugin/lib/vendor/propel-generator/classes/propel/engine/database/model/Table.php
===================================================================
--- lib/vendor/symfony/lib/plugins/sfPropelPlugin/lib/vendor/propel-generator/classes/propel/engine/database/model/Table.php	(revision 7888)
+++ lib/vendor/symfony/lib/plugins/sfPropelPlugin/lib/vendor/propel-generator/classes/propel/engine/database/model/Table.php	(working copy)
@@ -550,7 +550,7 @@
 	 */
 	public function getName()
 	{
-		return $this->name;
+		return 'q_'.$this->name;
 	}
 
 	/**
@@ -1148,4 +1148,14 @@
 		}
 		return $result;
 	}
+
+  public function getAttribute($name, $default = null)
+  {
+    if ($name == 'i18nTable')
+    {
+      return 'q_'.parent::getAttribute($name, $default);
+    }
+
+    return parent::getAttribute($name, $default);
+  }
 }
