Index: lib/vendor/symfony/lib/plugins/sfPropelPlugin/lib/propel/sfPropelDatabaseSchema.class.php
===================================================================
--- lib/vendor/symfony/lib/plugins/sfPropelPlugin/lib/propel/sfPropelDatabaseSchema.class.php	(revision 7939)
+++ lib/vendor/symfony/lib/plugins/sfPropelPlugin/lib/propel/sfPropelDatabaseSchema.class.php	(working copy)
@@ -362,6 +362,12 @@
           $this->setIfNotSet($this->database[$main_table]['_attributes'], 'isI18N', 1);
           $this->setIfNotSet($this->database[$main_table]['_attributes'], 'i18nTable', $i18n_table);
 
+	  // Add source_culture column to main table
+	  $this->setIfNotSet($this->database[$main_table], 'source_culture', array(
+	    'required' => true,
+	    'size' => 7,
+	    'type' => 'varchar'));
+
           // set id and culture columns for i18n table
           $this->setIfNotSet($this->database[$i18n_table], 'id', array(
             'type'             => 'integer',
@@ -401,7 +407,8 @@
           if ($column == 'created_at' || $column == 'updated_at')
           {
             // timestamp convention
-            $this->database[$table][$column]['type']= 'timestamp';
+            $this->database[$table][$column]['required'] = true;
+            $this->database[$table][$column]['type'] = 'timestamp';
           }
 
           if ($column == 'id')
@@ -434,7 +441,6 @@
               throw new sfException(sprintf('Unable to resolve foreign table for column "%s".', $column));
             }
           }
-          
         }
         else
         {
