Index: plugins/sfThumbnailPlugin/lib/sfImageMagickAdapter.class.php
===================================================================
--- plugins/sfThumbnailPlugin/lib/sfImageMagickAdapter.class.php	(revision 10558)
+++ plugins/sfThumbnailPlugin/lib/sfImageMagickAdapter.class.php	(working copy)
@@ -176,7 +176,8 @@
     $imgData = @getimagesize($image);
     if (!$imgData)
     {
-      exec($this->magickCommands['identify'].' '.escapeshellarg($image), $stdout, $retval);
+      $extract = $this->getExtract();
+      exec($this->magickCommands['identify'].' '.escapeshellarg($image).$extract, $stdout, $retval);
       if ($retval === 1)
       {
         throw new Exception('Image could not be identified.');
@@ -293,15 +294,7 @@
     }
 
     // extract images such as pages from a pdf doc
-    $extract = '';
-    if (isset($this->options['extract']) && is_int($this->options['extract']))
-    {
-      if ($this->options['extract'] > 0)
-      {
-        $this->options['extract']--;
-      }
-      $extract = '['.escapeshellarg($this->options['extract']).'] ';
-    }
+    $extract = $this->getExtract();
 
     $output = (is_null($thumbDest))?'-':$thumbDest;
     $output = (($mime = array_search($targetMime, $this->mimeMap))?$mime.':':'').$output;
@@ -328,5 +321,20 @@
   {
     return $this->sourceMime;
   }
+  
+  private function getExtract()
+  {
+    $extract = '';
+    if (isset($this->options['extract']) && is_int($this->options['extract']))
+    {
+      if ($this->options['extract'] > 0)
+      {
+        $this->options['extract']--;
+      }
+      $extract = '['.escapeshellarg($this->options['extract']).'] ';
+    }
+    
+    return $extract;
+  }
 
 }
