Index: lib/vendor/symfony/lib/yaml/sfYamlInline.class.php
===================================================================
--- lib/vendor/symfony/lib/yaml/sfYamlInline.class.php	(revision 9414)
+++ lib/vendor/symfony/lib/yaml/sfYamlInline.class.php	(working copy)
@@ -56,6 +56,8 @@
   {
     switch (true)
     {
+      case $value instanceof ReflectionCode:
+        return '<?php '.$value.' ?>';
       case is_resource($value):
         throw new InvalidArgumentException('Unable to dump PHP resources in a YAML file.');
       case is_object($value):
